app/controllers/pledgegroups_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PledgegroupsController < ApplicationController
+  def index
+    @pledgegroups = Pledgegroup.all
+  end
+
+  def new
+    @pledgegroup = Pledgegroup.new
+  end
+
+  
+  def create
+    @pledgegroup = Pledgegroup.new(pledgegroup_params)
+
+    if @pledgegroup.save
+      redirect_to pledgegroups_url
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+
+
+
+  # Get new pledgegroup parameters from form
+  private
+    def pledgegroup_params
+      @user ||= get_curr_user
+
+      par = params.require(:pledgegroup).permit(:budget)
+      par[:admins] = [@user]
+
+      # if checkbox checked, make is_public true
+      if params[:is_public]
+        par[:is_public] = true
+      end
+
+      par
```

Comment:
```suggestion
      return { budget: params[:pledgegroup][:budget], admins: [current_user], is_public: params[:is_public] }
```
Can do the same thing with a quick one-liner. OR even better - can get rid of this method altogether (since it's only called from one place). See review above. Would also allow you to get rid of the `get_curr_user` method entirely, too